tests/end2end/test_permission_deleting.py

```diff
@@ -23,21 +24,68 @@ def test_superuser_can_delete_existing_permission(superuser, permission, testapp
     res = form.submit().follow()
     # Clicks Permissions button
     res = res.click(_('Permissions'))
-    # Clicks Edit button on a permission
-    res = res.click(href=url_for('permission.edit', permission_id=permission.id))
     # Clicks Delete button on a permission
     permission_user_email = permission.user.email
     permission_collection_code = permission.collection.code
-    res = res.click(_('Delete Permission')).follow()
+    res = res.click(href=url_for('permission.delete', permission_id=permission.id) +
+                    '\?next=' + url_for('permission.home'))
     assert res.status_code == 200
+    assert _('Acknowledge Deletion') in res
+    assert _('Delete permissions for "%(username)s" on collection "%(code)s"?',
+             username=permission_user_email, code=permission_collection_code) in res
+    form = res.forms['deletePermissionForm']
+    form['acknowledged'] = 'y'
+    res = form.submit()
+    assert res.status_code == 302
+    assert url_for('permission.home') in res.location
+    res = res.follow()
+    # Permission was deleted, so number of permissions are 1 less than initial state
+    assert _('Successfully deleted permissions for "%(username)s" on collection "%(code)s".',
+             username=permission_user_email, code=permission_collection_code) in res
+    assert len(Permission.query.all()) == old_count - 1
+
+
+def test_cataloging_admin_can_delete_existing_permission(user, permission, superuser, testapp):
+    """Delete existing permission as cataloging admin for a collection."""
+    PermissionFactory(user=user, collection=permission.collection,
+                      cataloging_admin=True).save_as(superuser)
+    assert user.is_cataloging_admin_for(permission.collection) is True
+    old_count = len(Permission.query.all())
+    permission_user_email = permission.user.email
+    permission_collection_code = permission.collection.code
+    # Goes to homepage
+    res = testapp.get('/')
+    # Fills out login form
+    form = res.forms['loginForm']
+    form['username'] = user.email
+    form['password'] = 'myPrecious'
+    # Submits
+    res = form.submit().follow()
+
+    # We see no Permissions button
+    assert res.lxml.xpath("//a[contains(@text,'{0}')]".format(_('Permissions'))) == []
+
+    # Try to go there directly
+    testapp.get('/permissions/', status=403)
+
+    # Try to delete a specific permission directly
+    res = testapp.get(url_for('permission.delete', permission_id=permission.id))
```

Comment:
I would prefer if we test the whole flow required for a cataloging admin to find the permission to delete. Should we go through the profile page?

Reply:
Right now there's no way for a user to find any "delete permission" links. Only way would be if they read the code and made some educated guesses as to what URL to type in.

In the branch I'm currently working on I'll be adding edit/delete buttons to the collection view list of privileges, so my plan is to update this test accordingly once the whole flow is implemented!

Reply:
:+1: